pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/errorprone/xml/UselessCondition.xml

```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<test-data
+        xmlns="http://pmd.sourceforge.net/rule-tests"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://pmd.sourceforge.net/rule-tests https://pmd.github.io/schema/rule-tests_1_1_0.xsd">
+
+    <test-code>
+        <description>Useless if</description>
+        <expected-problems>1</expected-problems>
+        <code><![CDATA[
+        class Test {
+            int method1() {
+                if (Math.random() > 0.5) {
+                    return 1;
+                } else {
+                    return 1;
+                }
+            }
+        }
+        ]]></code>
+    </test-code>
+    <test-code>
+        <description>Useless conditional expression</description>
+        <expected-problems>1</expected-problems>
+        <code><![CDATA[
+        class Test {
```

Comment:
Can we add a test for eg
- empty branches like `if(something()); else {}`
- comments eg `if (something()) { /*foo*/ something(); } else { something(); }`
I'm genuinely not sure how tokenEquals treats those situations